echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/controllers/WebhooksController.groovy

```diff
@@ -61,11 +57,26 @@ class WebhooksController {
       }
     }
 
-    log.info("Webhook ${source}:${type}:${event.content}")
+    log.info("Webhook ${type}:${source}:${event.content}")
 
     if (sendEvent) {
       propagator.processEvent(event)
     }
   }
 
+  @RequestMapping(value = '/webhooks/{type}', method = RequestMethod.POST)
+  void forwardEvent(@PathVariable String type, @RequestBody Map postedEvent) {
+    Event event = new Event()
+    event.details = new Metadata()
+    event.details.type = type
+    event.content = postedEvent
+
+    if (event.content.source != null) {
+      event.details.source = event.content.source;
+    }
+
+    log.info("Webhook ${type}:${event.details.source}:${event.content}")
+
+    propagator.processEvent(event)
+  }
```

Comment:
Can you explain what this controller method does? When I look at the method above, it seems like you get the same behavior by posting to the same URL with a trailing slash, e.g. `http://echo/webhooks/teamcity/` instead if `http://echo/webhooks/teamcity`

If we don't need this method, we should remove it; otherwise, we need to expose it through an endpoint in Gate.